x-pack/plugins/apm/server/routes/services/get_service_transaction_groups_alerts.ts

```diff
@@ -117,8 +117,9 @@ export async function getServiceTranactionGroupsAlerts({
 
   const response = await apmAlertsClient.search(params);
 
-  const { buckets } = response.aggregations
-    ?.transaction_groups as ServiceTransactionGroupAlertsAggResponse;
+  const { buckets } = (response?.aggregations?.transaction_groups ?? {
+    buckets: [],
+  }) as ServiceTransactionGroupAlertsAggResponse;
```

Comment:
Is there an issue with the types? This should be possible and will be type safe:

```suggestion
  const buckets = response.aggregations?.transaction_groups?.buckets ?? [];
```

Reply:
Alert client search method:
https://github.com/elastic/kibana/blob/933d97ff6d6c90976e947e088708df4127954a87/x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts#L319-L324

APM search method (response type inference)
https://github.com/elastic/kibana/blob/933d97ff6d6c90976e947e088708df4127954a87/x-pack/plugins/apm/server/routes/diagnostics/create_typed_es_client.ts#L18-L22

Reply:
yes, by default the returned aggregations are of type `AggregationsAggregate` which do not have `buckets`. We need to cast them

Reply:
Can you fix the types returned by the alerting client. We shouldn't have to cast when using a client. That's what allowed this error to get in.

Reply:
@sqren I tried adjusting the types returned by the Alerting Client. This seems to be a bit complicated and cascading as well given the scope. I understand the whole issue happened due to type casting at the 1st place, but this would need a detailed look by the Alerting team may be.

Reply:
> but this would need a detailed look by the Alerting team may be.

Could we just override the return type in our own alerting client?

Reply:
@achyutjhunjhunwala Try updating the alerting client in `get_apm_alerts_client.ts` with this:

```ts
  type RequiredParams = ESSearchRequest & {
    size: number;
    track_total_hits: boolean | number;
  };

  return {
    search<TParams extends RequiredParams>(
      opts: TParams
    ): Promise<InferSearchResponseOf<ParsedTechnicalFields, TParams>> {
      return alertsClient.find({
        ...opts,
        index: apmAlertsIndices.join(','),
      }) as Promise<any>;
    },
  };
```

You will also need to update the types for `track_total_hits` in `x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts` to accept numbers:

```
track_total_hits?: boolean | number | undefined;
```

Reply:
Good idea, i have updated our Alerting client accordingly. I didn't touch the `rule_registry` code to avoid triggering additional code reviewers. We don't need the `track_total_hits` unless we also want to update our queries as currently we don't use it?